src/controls/DateTimeControl.tsx

```diff
@@ -0,0 +1,75 @@
+import { memo } from "react"
+import type { ControlProps as JSFControlProps } from "@jsonforms/core"
+import { withJsonFormsControlProps } from "@jsonforms/react"
+import { DatePicker, type DatePickerProps, Form } from "antd"
+import type { Rule } from "antd/es/form"
+import dayjs from "dayjs"
+
+import {
+  ControlUISchema,
+  DateTimeControlOptions,
+  isDateTimeControlOptions,
+} from "../ui-schema"
+
+type ControlProps = Omit<JSFControlProps, "uischema"> & {
+  uischema: ControlUISchema<unknown> | JSFControlProps["uischema"]
+}
+// initialize once
+const DEFAULT_PROPS: DateTimeControlOptions = {
+  format: { format: "YYYY-MM-DD HH:mm:ss", type: "mask" },
+} as const
+
+function getProps(options: unknown): DateTimeControlOptions {
+  if (isDateTimeControlOptions(options)) {
+    return options
+  }
+  return DEFAULT_PROPS
+}
+
+export function DateTimeControl({
+  handleChange,
+  path,
+  label,
+  id,
+  required,
+  schema,
+  uischema,
+  visible,
+}: ControlProps) {
+  if (!visible) return null
+
+  const initialValue =
+    typeof schema.default === "string" ? dayjs(schema.default) : undefined
```

Comment:
Using our own instance of dayjs isn't ideal (or even using a hardcoded timezone library since AntD lets you generate time-based ui controls with your own library) and may cause problems in any app that has a configurable timezone, but I think it's ok for us to use it here since AntD does a pretty poor job making the timezone configurable (it doesn't, and it overrides any default you set on dayjs 😕). If we wanted to support configurable timezones, we'd have to use the generatePicker function and probably accept a datetimelibraryconfig object (inside of jsonforms's config) that contains library-specific details that we could use to generate the AntD DateTime component.
https://github.com/ant-design/ant-design/issues/47490#issuecomment-1970505978